src/main.rs

```diff
@@ -139,8 +140,9 @@ fn fs_prep(pid: unistd::Pid) -> String {
     rootfs
 }
 
-fn net_prep(/* pid, link, ip */) {
-    // network setup
+fn net_prep(ip: String, pid: u32) {
```

Comment:
should the type of `pid` be of type `nix::unistd::Pid` or `libc::pid_t` perhaps?

Reply:
[pub fn setns_by_pid(mut self: Self, pid: u32) -> Self](https://docs.rs/rtnetlink/0.7.0/rtnetlink/struct.LinkSetRequest.html#method.setns_by_pid)
this method take u32, so it need convert type at the end.

Reply:
I think `libc::pid_t` declaration is `type pid_t = u32`

Reply:
it's i32

Reply:
Got it